scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -411,6 +411,16 @@ class FormatOps(
     template.tokens.find(x => x.is[T.LeftBrace] && owners(x) == template)
   }
 
+  def templateDerivesOrCurly(template: Template): Option[Token] = {
+    val tokens = template.tokens
+    val index = tokens.indexWhere(x =>
+      (x.is[soft.KwDerives] || x.is[T.LeftBrace]) && owners(x) == template
+    )
+    if (index < 0) None
+    else if (tokens(index).is[soft.KwDerives]) Some(tokens(index - 1))
```

Comment:
and if `index == 0`? :)

Reply:
Changed to `index <= 0` - in reality this should never happen (hopefully)